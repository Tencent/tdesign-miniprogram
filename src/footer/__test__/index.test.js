import path from 'path';
import simulate from 'miniprogram-simulate';

describe('Footer', () => {
  const footer = load(path.resolve(__dirname, `../footer`));

  it(`: style && customStyle`, async () => {
    const id = simulate.load({
      template: `<t-footer class="footer" style="{{style}}" customStyle="{{customStyle}}"></t-footer>`,
      usingComponents: {
        't-footer': footer,
      },
      data: {
        style: 'color: red',
        customStyle: 'font-size: 9px',
      },
    });
    const comp = simulate.render(id);
    comp.attach(document.createElement('parent-wrapper'));
    const $footer = comp.querySelector('.footer >>> .t-footer');
    // expect(comp.toJSON()).toMatchSnapshot();
    if (VIRTUAL_HOST) {
      expect($footer.dom.getAttribute('style').includes(`${comp.data.style}; ${comp.data.customStyle}`)).toBeTruthy();
    } else {
      expect($footer.dom.getAttribute('style').includes(`${comp.data.customStyle}`)).toBeTruthy();
    }
  });

  it(':base', () => {
    const id = simulate.load({
      template: `<t-footer></t-footer>`,
      usingComponents: {
        't-footer': footer,
      },
    });
    const comp = simulate.render(id);
    comp.attach(document.createElement('parent-wrapper'));

    expect(comp.toJSON()).toMatchSnapshot();
  });
});
