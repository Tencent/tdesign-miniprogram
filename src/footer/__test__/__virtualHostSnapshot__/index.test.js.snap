// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer :base 1`] = `
<main>
  <wx-view
    class="t-footer class t-class"
    style=""
  >
    <wx-view
      class="t-footer__text"
    >
      
    </wx-view>
  </wx-view>
</main>
`;
